### ISSUE TITLE
Semantic Versions

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [netty/netty#1269]
Since Netty is now converted to OSGI,
I suggest to make the next step:
introduce Semantic Versions.

Basic idea is described here:
http://www.aqute.biz/Bnd/Versioning
http://www.osgi.org/wiki/uploads/Links/SemanticVersioning.pdf

And elaborated here in how to apply it to Maven:
https://github.com/barchart/barchart-version-tester/wiki/Version-Policy

Here is a tester project:
https://github.com/barchart/barchart-version-tester

Bottom line: with help of
- maven bundle plugin 2.4.0 / bnd 2.0.0
- aries version check plugin 0.1.0
- RFC 197 OSGi Package and Type Annotation / bnd annotatins

Netty can actually enforece and guarantee binary and semantic
compatibility between its own releases.
